arci-ros2/src/ros2_control.rs

```diff
@@ -0,0 +1,153 @@
+use std::{
+    sync::{
+        atomic::{AtomicBool, Ordering},
+        Arc,
+    },
+    time::Duration,
+};
+
+use arci::*;
+use futures::stream::StreamExt;
+use parking_lot::Mutex;
+use r2r::{
+    builtin_interfaces::{msg as builtin_msg, msg::Time},
+    control_msgs::action::FollowJointTrajectory,
+    std_msgs::msg::Header,
+    trajectory_msgs::msg as trajectory_msg,
+};
+use serde::{Deserialize, Serialize};
+
+/// Implement arci::JointTrajectoryClient for ROS2
+pub struct Ros2ControlClient {
+    action_client: r2r::ActionClient<FollowJointTrajectory::Action>,
+    /// r2r::Node to handle the action
+    node: Arc<Mutex<r2r::Node>>,
+    joint_names: Vec<String>,
+}
+
+// TODO:
+unsafe impl Sync for Ros2ControlClient {}
+
+impl Ros2ControlClient {
+    /// Create instance from ROS2 context and names of action and joints.
+    pub fn new(ctx: r2r::Context, action_name: &str, joint_names: Vec<String>) -> Self {
+        // TODO: Use unique name
+        let mut node = r2r::Node::create(ctx, "nav2_node", "arci_ros2").unwrap();
```

Comment:
'nav2_node' means for ROS2 node for navigation2.
let's change to ros2_control_node or something.
Anyway, I think we need to make it possible to set the value. (Add an argument? (but navigation client also need that.))